src/fn/create-users.js

```diff
@@ -1,52 +1,92 @@
 const fs = require('../lib/sync-fs');
-const info = require('../lib/log').info;
+const csvParse = require('csv-parse/lib/sync');
+const { info, warn, error } = require('../lib/log');
 const request = require('request-promise-native');
+const readline = require('readline-sync');
 
-module.exports = (projectDir, couchUrl) => {
-  if(!couchUrl) throw new Error('Server URL must be defined to use this function.');
-  const instanceUrl = couchUrl.replace(/\/medic$/, '');
-
-  return Promise.resolve()
-    .then(() => {
-      const csvPath = `${projectDir}/users.csv`;
-
-      if(!fs.exists(csvPath)) throw new Error(`User csv file not found at ${csvPath}`);
-
-      const { cols, rows } = fs.readCsv(csvPath);
-      const usernameIndex = cols.indexOf('username');
-      const passwordIndex = cols.indexOf('password');
-      const rolesIndex = cols.indexOf('roles');
-      const placeIdIndex = cols.indexOf('place');
-      const contactIndex = cols.indexOf('contact');
-
-      return rows.reduce((promiseChain, row) => {
-        const username = row[usernameIndex];
-        const password = row[passwordIndex];
-        const roles     = row[rolesIndex].split(':');
-        const contact = contactIndex === -1 ? prefixedProperties(cols, row, 'contact.') : row[contactIndex];
-        const place = placeIdIndex === -1 ? prefixedProperties(cols, row, 'place.') : row[placeIdIndex];
-        const requestObject = { username, password, roles, place, contact };
-
-        return promiseChain
-          .then(() => {
-            info('Creating user', username);
-            return request({
-              uri: `${instanceUrl}/api/v1/users`,
-              method: 'POST',
-              json: true,
-              body: requestObject,
-            });
-          });
-      }, Promise.resolve());
+const nestPrefixedProperties = (obj, name) => {
+  const nested = {};
+  const prefix = `${name}.`;
+  Object
+    .keys(obj)
+    .filter(key => key.startsWith(prefix))
+    .forEach(key => {
+      nested[key.substring(prefix.length)] = obj[key];
+      delete obj[key];
     });
+  return obj[name] || nested;
 };
 
-function prefixedProperties (cols, row, prefix) {
-  const indices = {};
-  cols.forEach(col => {
-    if (col.startsWith(prefix)) {
-      indices[col.substring(prefix.length)] = row[cols.indexOf(col)];
-    }
+const getUsersData = (csvData) => {
```

Comment:
A more specific verb here would more clearly communicate the function's behavior. Maybe `parseUsersData`?